### ISSUE TITLE
FreeRTOS tud_task() exit

### ISSUE THREAD

**Issue by nitrotm** [hathach/tinyusb#408]
Currently `tud_task` is waiting on a queue indefinitely without any option to abort the loop. A solution would be to add an 'exit' message that can be sent to stop the loop.

Example:

```
void tud_stop(uint8_t rhport)
{
  dcd_event_t event = { .rhport = rhport, .event_id = DCD_EVENT_EXIT, };

  dcd_event_handler(&event, false);
}
```

and handle the event in `tud_task`.

**Comment by hathach** [comment]
currently, I don't see an reason to do this. 

PS: Next time please use the the feature template whenever you requested an new feature.

**Comment by nitrotm** [comment]
@hathach I did choose 'feature' template but the label was not applied when I submitted. Maybe I should have kept the template text within?

My use-case is based on FreeRTOS and I spawn a dedicated task to handle usb core events in `tud_task`. But I also want to offer the option to turn off the usb stack (so being able to start or stop it dynamically). Therefore I implemented the `DCD_EVENT_EXIT` described above to signal the task waiting on the semaphore to wake-up and return.
